ethportal-peertest/src/cli.rs

```diff
@@ -1,68 +1,81 @@
-use std::{env, ffi::OsString};
-use structopt::StructOpt;
+use clap::Parser;
 use trin_types::cli::{
     DEFAULT_WEB3_HTTP_ADDRESS as DEFAULT_TARGET_HTTP_ADDRESS,
     DEFAULT_WEB3_IPC_PATH as DEFAULT_TARGET_IPC_PATH,
 };
 
 const DEFAULT_LISTEN_PORT: &str = "9876";
 
-#[derive(StructOpt, Debug, PartialEq, Eq, Clone)]
-#[structopt(
+#[derive(Parser, Debug, PartialEq, Eq, Clone)]
+#[command(
     name = "ethportal-peertest",
     version = "0.0.1",
     about = "Testing framework for portal network peer-to-peer network calls"
 )]
 pub struct PeertestConfig {
-    #[structopt(
-        default_value(DEFAULT_LISTEN_PORT),
-        short = "p",
+    #[arg(
+        default_value = DEFAULT_LISTEN_PORT,
+        short = 'p',
         long = "listen-port",
         help = "The UDP port to listen on."
     )]
     pub listen_port: u16,
 
-    #[structopt(
+    #[arg(
         default_value = "ipc",
-        possible_values(&["http", "ipc"]),
+        value_parser = ["http", "ipc"],
```

Comment:
I found `value_parser` was not for possible values. It will deal with the input string and compare with it. The above code will get an error when the input is "ipc", so this should be fixed.
I think a good way is to use `enum` to list all the possible values and `derive` by `clap::ValueEnum`.